eos-live-boot-overlayfs-setup

```diff
@@ -47,18 +44,41 @@ setup_ostree_flatpak_overlay() {
     # via the /var/lib/flatpak symlink.
 }
 
-# Everything but /sysroot/{ostree,flatpak} is pretty straightforward:
-overlay_dirs="bin boot endless home lib opt ostree root sbin srv sysroot/home var"
-for dir in $overlay_dirs $EXTRA_PATHS; do
-    [ -d /$dir ] || continue
-    [ -d /run/eos-live/$dir ] && continue
+setup_overlay() {
+    local dir=$1
+    [ -d /$dir ] || return
+    [ -d /run/eos-live/$dir ] && return
     # If the directory is a symlink, assume it's pointing to a location
     # covered by another top level overlay
-    [ -L /$dir ] && continue
+    [ -L /$dir ] && return
     mkdir -p /run/eos-live/$dir /run/eos-live/$dir-workdir
     mount -t overlay -o \
         rw,upperdir=/run/eos-live/$dir,lowerdir=/$dir,workdir=/run/eos-live/$dir-workdir \
         eos-live-$dir /$dir
+}
+
+setup_overlay etc
+
+# If we booted and systemd was unable to find or create a machine-id,
+# it will create one at /run/machine-id and bind mount it at /etc/machine-id.
+# But now that we just setup an overlay at /etc, we lost that bind mount, so
```

Comment:
Huh, I did not expect that mounting an overlay over `/etc` with lowerdir `/etc` would lose the bind-mount within `/etc`.